src/h5web/vis-packs/core/scales.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  scaleLinear,
+  scaleLog,
+  scaleSymlog,
+  scaleSqrt,
+  PickScaleConfig,
+} from '@visx/scale';
+import type { IconType } from 'react-icons/lib';
+import type { AxisScale, Domain } from './models';
+import { ScaleType } from './models';
+import { MdSort, MdFilterList } from 'react-icons/md';
+import SqrtIcon from '../../toolbar/controls/ScaleSelector/SqrtIcon';
+import MdGraphicEqRotated from '../../toolbar/controls/ScaleSelector/MdGraphicEqRotated';
+
+interface H5WebScale {
+  createScale: (
+    config: Omit<PickScaleConfig<ScaleType, number>, 'type'>
+  ) => AxisScale;
+  Icon: IconType;
+  label: string;
+  scalingFunction: (val: number) => number;
+  supportedDomain: Domain;
+
+  supportsDomain: (domain: Domain) => boolean;
+  shaderImplementation: string;
+}
+
+const linearScale: H5WebScale = {
+  createScale: (config) => scaleLinear<number>(config),
+  Icon: MdSort,
+  label: 'Linear',
+  scalingFunction: (val) => val,
+  supportedDomain: [-Number.MAX_VALUE / 2, Number.MAX_VALUE / 2],
+
+  supportsDomain: () => true,
+  shaderImplementation: `
+  float scale(float value) {
+    return oneOverRange * (value - min); // Linear
+  }
+
+  bool isSupported(float value) {
+    return true;
+  }`,
```

Comment:
I tried to keep the `shaderImplementation` to the bare minimum. Unfortunately, this implies some implicit coupling as one needs to know which variables are defined and which functions needs to be defined by the shader. 

Open to suggestions on this.

Reply:
It's the downside of having a centralised `Record`: it becomes difficult to know where all these properties (`supportedDomain`, `supportsDomain`, etc.) are used and what they are used for. All the functions are disconnected from the code that uses them.

With the vis def records `CORE_VIS` and `NEXUS_VIS`, this pattern works because the records are used in very few places, and most of the properties hold React components and providers.

Here, because the scales are used in so many places in the codebase, and because some of the things stored in the record are highly coupled with the caller code (e.g. `scalingFunction`, `supportsDomain`, `shaderImplementation`), it can be a bit difficult to understand.